doc/specs/stdlib_bitsets.md

```diff
@@ -0,0 +1,2006 @@
+        call set1 % write_bitset(unit, advance='no')
+        call set0 % write_bitset(unit)
+        close( unit )
+        open( newunit=unit, file='test.txt', status='old', &
+            form='formatted', action='read' )
+        call set3 % read_bitset(unit, advance='no')
+        call set4 % read_bitset(unit, advance='no')
+        call set5 % read_bitset(unit)
+        if ( set3 == set0 .and. set4 == set1 .and. set5 == set2 ) then
+            write(*,*) WRITE_BITSET to READ_BITSET through unit worked.'
+        end if
+    end program demo_read_bitset
+```
+
+### `set` - sets a sequence of one or more bits to 1.
+
+#### Status
+
+Experimental
+
+#### Description
+
+Sets a sequence of one or more bits in `self` to 1.
+
+* If `start_pos` and `end_pos` are absent sets the bit at position
+`pos` in `self` to 1.
+
+* If `start_pos` and `end_pos` are present with `end_pos >= start_pos`
+set the bits at positions from `start_pos` to `end_pos` in `self` to 1.
+
+* If `start_pos` and `end_pos` are present with `end_pos < start_pos`
+`self` is unchanged.
+
+* Positions outside the range 0 to `bits(self)` are ignored.
+
+
+#### Syntax
+
+`call self % [[bitset_type(class):set(bound)]] (POS)`
+
+or
+
+`call self % [[bitset_type(class):set(bound)]] (START_POS, END_POS)`
+
+#### Class
+
+Elemental subroutine
+
+#### Arguments
+
+`self`: shall be a scalar variable of class `bitset_type`. It is an
+  `intent(inout)` argument.
+
+`pos` (optional): shall be a scalar integer expression of kind
+`bits_kind`. It is an `intent(in)` argument.
+
+`start_pos` (optional): shall be a scalar integer expression of kind
+`bits_kind`. It is an `intent(in)` argument.
+
+`end_pos` (optional): shall be a scalar integer expression of kind
+`bits_kind`. It is an `intent(in)` argument.
+
+#### Example
+
+```fortran
+    program demo_set
+        use stdlib_bitsets
+        type(bitset_large) :: set0
+        call set0 % init(166)
+        if ( set0 % none() ) write(*,*) 'SET0 is properly initialized.'
+        call set0 % set(165)
+        if ( set0 % test(165) ) write(*,*) 'Bit 165 is set.'
+        call set0 % set(0,164)
+        if ( set0 % all() ) write(*,*) 'All bits are set.'
+    end program demo_set
+```
+
+### `test` - determine whether a bit is set
+
+#### Status
+
+Experimental
+
+#### Descriptions
+
+Determine whether the bit at position `pos` is set to 1 in `self`.
+
+
+#### Syntax
+
+`result = self % [[bitset_type(class):test(bound)]](pos)`
+
+#### Class
+
+Elemental function.
+
+#### Arguments
+
+`self`: shall be a scalar expression of class `bitset_type`. It is an
+`intent(in)` argument.
+
+`pos`: shall be a scalar integer expression of kind `bits_kind`. It is
+an `intent(in)` argument.
+
+#### Result character
+
+The result is a default logical scalar.
+
+#### Result value
+
+The result is `.true.` if the bit at `pos`, in `self` is set,
+otherwise it is `.false.`. If `pos` is outside the range
+`0... bits(self)-1` the result is `.false.`.
+
```

Comment:
" If `pos` is outside the range ... `.false.`.": does this make sense? As a user, I would prefer that the program stops, or at least, returns something corresponding to `NaN`